### PR TITLE
VIH-8380 Staff member or judge joins hearing from consultation room

### PR DESCRIPTION
**Pull request by robertbennett1998**
### JIRA link (if applicable) ###
[VIH-8380](https://tools.hmcts.net/jira/browse/VIH-8380)

### Change description ###
- Added a button to consultation room to allow judge/staff member to join the hearing from the consultation room when the hearing is in session.

**Does this PR introduce a breaking change?** (check one with "x")
```
[ ] Yes
[x] No
```

**Notes
- Still waiting on a change on Kinlys side to stop the Judge/SM been moved from consultation to waiting room on hearing start.

### CODE REVIEW

File: VideoWeb/VideoWeb/ClientApp/src/app/waiting-space/waiting-room-shared/waiting-room-base.component.ts [hunk 1]
```diff
@@ -1123,7 +1123,7 @@ export abstract class WaitingRoomBaseDirective {
     }
 
     shouldCurrentUserJoinHearing(): boolean {
-        return !this.isHost() || this.dualHostHasSignalledToJoinHearing;
+        return this.isHost() ? this.dualHostHasSignalledToJoinHearing || this.participant.status === ParticipantStatus.InHearing : true;
```
Comment (PrasannaDommalapati) [inline VideoWeb/VideoWeb/ClientApp/src/app/waiting-space/waiting-room-shared/waiting-room-base.component.ts:1126]: have you tried the ejud panel member in the hearing, beacuse Ejud panel members have role Judge

**Review by PrasannaDommalapati — DISMISSED** [review]
LGTM

File: VideoWeb/VideoWeb/ClientApp/src/app/waiting-space/waiting-room-shared/waiting-room-base.component.ts (hunk 1 above)
Comment (robertbennett1998) [reply VideoWeb/VideoWeb/ClientApp/src/app/waiting-space/waiting-room-shared/waiting-room-base.component.ts:1126]: No but that would then give them access to the judge view right?
Is this expected? I will have a look now

File: VideoWeb/VideoWeb/ClientApp/src/app/waiting-space/private-consultation-room-controls/private-consultation-room-controls.component.scss [hunk 2]
```diff
@@ -111,6 +111,14 @@
   }
 }
 
+.small-button.green {
+  color: govuk-colour('white');
+  background: govuk-colour('green');
+  &:hover {
+    background: govuk-colour('green');
+  }
```
Comment (FarronF) [inline VideoWeb/VideoWeb/ClientApp/src/app/waiting-space/private-consultation-room-controls/private-consultation-room-controls.component.scss:119]: Is this to override a hover style from another class?

**Comment by FarronF** [comment]
This looks ok. But what's with the huge amount of changes in the i18n files? Is it just indentation changes that git is struggling to show correctly?

File: VideoWeb/VideoWeb/ClientApp/src/app/waiting-space/waiting-room-shared/waiting-room-base.component.ts (hunk 1 above)
Comment (robertbennett1998) [reply VideoWeb/VideoWeb/ClientApp/src/app/waiting-space/waiting-room-shared/waiting-room-base.component.ts:1126]: Checked this and the isHost works correctly for ejud

image

**Review by FarronF — APPROVED** [review]

File: VideoWeb/VideoWeb/ClientApp/src/app/waiting-space/private-consultation-room-controls/private-consultation-room-controls.component.scss (hunk 2 above)
Comment (robertbennett1998) [reply VideoWeb/VideoWeb/ClientApp/src/app/waiting-space/private-consultation-room-controls/private-consultation-room-controls.component.scss]: I have removed as it was not necessary

**Review by FarronF — APPROVED** [review]

**Review by PrasannaDommalapati — APPROVED** [review]

**Review by FarronF — APPROVED** [review]

**Pull request merged**